File: vvd-extractor/src/main/java/mikufan/cx/vvd/extractor/config/EnvironmentConfig.java
package mikufan.cx.vvd.extractor.config;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;
import lombok.experimental.FieldDefaults;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.NotBlank;

/**
 * @author CX无敌
 * @date 2020-12-28
 */
@ConfigurationProperties(prefix = "config.environment")
@ConstructorBinding
@Validated
@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
@AllArgsConstructor
@Getter
@ToString
public class EnvironmentConfig {

  @NotBlank String pythonLaunchCmd;
  @NotBlank String ffmpegLaunchCmd;

}

File: vvd-extractor/src/main/kotlin/mikufan/cx/vvd/extractor/config/EnvironmentConfig.kt
package mikufan.cx.vvd.extractor.config

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

/**
 * @date 2022-06-12
 * @author CX无敌
 */
@ConfigurationProperties("config.environment")
@Validated
data class EnvironmentConfig(
  @field:NotEmpty val pythonLaunchCmd: List<@NotBlank String>,
  @field:NotEmpty val ffmpegLaunchCmd: List<@NotBlank String>,
  @field:NotEmpty val mediainfoLaunchCmd: List<@NotBlank String>,
)
